controller_manager:
  ros__parameters:
    update_rate: 125  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    left_hand:
      type: joint_trajectory_controller/JointTrajectoryController

    right_hand:
      type: joint_trajectory_controller/JointTrajectoryController


left_hand:
  ros__parameters:
    allow_partial_joints_goal: True
    joints:
      - Left_Hand_Thumb_Flexion
      - Left_Hand_Thumb_Opposition
      - Left_Hand_Index_Finger_Distal
      - Left_Hand_Index_Finger_Proximal
      - Left_Hand_Middle_Finger_Distal
      - Left_Hand_Middle_Finger_Proximal
      - Left_Hand_Ring_Finger
      - Left_Hand_Pinky
      - Left_Hand_Finger_Spread
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

right_hand:
  ros__parameters:
    allow_partial_joints_goal: True
    joints:
      - Right_Hand_Thumb_Flexion
      - Right_Hand_Thumb_Opposition
      - Right_Hand_Index_Finger_Distal
      - Right_Hand_Index_Finger_Proximal
      - Right_Hand_Middle_Finger_Distal
      - Right_Hand_Middle_Finger_Proximal
      - Right_Hand_Ring_Finger
      - Right_Hand_Pinky
      - Right_Hand_Finger_Spread
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
